### PR TITLE
fix: add variable for customer managed encryption key for cluster autoscaling defaults

### PR DESCRIPTION
**Pull request by felix-lecuyer**
Fix issue #1566 . 

Add a new input variable for beta modules to input a customer managed encryption key used in cluster_autoscaling defaults. 

Add some examples on how to configure an autopilot and a node pool cluster with customer managed encryption key. 

Purpose of this change: be able to use the module in a project where the organization policy constraints/gcp.restrictNonCmekServices is active (e.g. in a sovereign folder).

### CODE REVIEW

File: autogen/main/variables.tf.tmpl [hunk 1]
```diff
@@ -753,3 +753,11 @@ variable "enable_identity_service" {
 }
   {% endif %}
 {% endif %}
+
+{% if beta_cluster %}
```
Comment (bharathkkb) [inline autogen/main/variables.tf.tmpl:757]: Any reason for restricting to beta? I see it available in the GA provider

https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4310-aug-1-2022

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @felix-lecuyer 
Could we showcase this feature in an existing example rather than adding a new example? We also try to keep the examples self contained so it would be good to spin up a sample key in the example itself rather than getting it as an input.

File: autogen/main/variables.tf.tmpl (hunk 1 above)
Comment (felix-lecuyer) [reply autogen/main/variables.tf.tmpl]: Hi Barath, 
Thanks for the feedback. No particular reason so I removed the restriction to beta in my last commit..

**Comment by felix-lecuyer** [comment]
Hi @bharathkkb, 
I managed to take into account your comments - sorry for the delay.
Looking forward to a new review. 

Many thanks.
Felix

**Comment by dangarthwaite** [comment]
Hi @bharathkkb, we really need this to comply with gcp.restrictNonCmekServices.  How can we get this merged?

**Pull request closed without merging**